src/frontend/src/optimizer/plan_node/logical_scan.rs

```diff
@@ -152,6 +164,7 @@ impl LogicalScan {
             new_required_col_idx,
             index.clone(),
             vec![],
+            self.distribution_keys.clone(),
```

Comment:
Index's distribution key is different from its original MV. Distribution keys should be stored in TableDesc.